airbyte-integrations/connectors/source-google-sheets/google_sheets_source/spec.json

```diff
@@ -4,18 +4,81 @@
     "$schema": "http://json-schema.org/draft-07/schema#",
     "title": "Stripe Source Spec",
     "type": "object",
-    "required": ["spreadsheet_id", "credentials_json"],
-    "additionalProperties": false,
+    "required": ["spreadsheet_id"],
+    "additionalProperties": true,
     "properties": {
       "spreadsheet_id": {
         "type": "string",
         "description": "The ID of the spreadsheet to be replicated."
       },
-      "credentials_json": {
-        "type": "string",
-        "description": "The contents of the JSON service account key. See the <a href=\"https://docs.airbyte.io/integrations/sources/google-sheets\">docs</a> for more information on how to generate this key.",
-        "airbyte_secret": true
+      "credentials": {
+        "type": "object",
+        "oneOf": [
+          {
+            "title": "Authenticate via Google (Oauth)",
+            "type": "object",
+            "required": [
+              "auth_type",
+              "client_id",
+              "client_secret",
+              "refresh_token"
+            ],
+            "properties": {
+              "auth_type": {
```

Comment:
I think, auth_type shouldn't be part of specification. It should be recognized in code.

Reply:
@vitaliizazmic this practice is used in the others connectors, could you point to some code with a new approach?